packages/datadog-instrumentations/src/graphql.js

```diff
@@ -337,6 +351,33 @@ function wrapFieldType (field) {
   wrapFields(unwrappedType)
 }
 
+function getResolverInfo (info, args) {
+  let resolverInfo = null
+  const resolverVars = {}
+
+  if (args && Object.keys(args).length) {
+    Object.assign(resolverVars, args)
+  }
+
+  const directives = info.fieldNodes[0].directives
+  for (const directive of directives) {
+    const argList = {}
+    for (const argument of directive['arguments']) {
+      argList[argument.name.value] = argument.value.value
+    }
+
+    if (Object.keys(argList).length) {
+      resolverVars[directive.name.value] = argList
+    }
+  }
+
+  if (Object.keys(resolverVars).length) {
+    resolverInfo = { [info.fieldName]: resolverVars }
+  }
+
+  return resolverInfo
+}
```

Comment:
It was intentional that this existed on the plugin side previously. The instrumentation should be concerned _only_ with forwarding the data it has and should avoid constructing new data structures, especially ones which are built around the opinions of a particular product.

We _intend_ for the instrumentations to gradually be copied into the libraries themselves, which means we shouldn't have _our own_ logic like this in the instrumentation if it can be avoided.

Reply:
Yep, the issue is that the previous implementation was only for monitoring and we didn't care about whether the resolver was executed or not. In this case we need to avoid the resolver execution so we need that information to be sent before it's executed. I moved it because the other solutions didn't seem quite good at the moment: one was to create an additional channel in the plugin to send the information to the appsec module and the other one was to move `getResolverInfo`to the appsec module, coupling the appsec implementation with the graphql one.

WDYT? the channel option would be acceptable?

Reply:
The channel is what we'll need to do anyway when we try to upstream this, so I'd say that's the way to go.

Reply:
OK, it's already done in the last commit